DKDBManager/NSManagedObject+DKDBManager.h

```diff
@@ -18,32 +38,308 @@ typedef NS_ENUM(NSInteger, DKDBManagedObjectState) {
+ * Depending on the returned predicate the manager will update an existing entity or will create a new one.
+ *
+ * If returns nil then will only take the first entity created (if any) and update it. By doing so only ONE entity will ever be created.`
+ *
+ * If returns a `false predicate` then a new entity will always be created.
+ *
+ * Otherwise use the entity found by the predicate.
+ * The predicate should be created depending on the parameter: `dictionary`.
+ *
+ * For example, your entity `Book` could be fetch through its `releaseDate` and `title` attributes.
+ *
+ * @return A NSPredicate object to find/fetch an entity in the local database.
+ */
+ + (NSPredicate *)primaryPredicateWithDictionary:(NSDictionary *)dictionary;
+
+/**
+ * @brief Returns all entities of the current class model.
+ *
+ * @discussion The values are sorted by the default sorting attribute.
+ *
+ * @see + (NSString *)sortingAttributeName;
+ *
+ * @return A NSArray object containing all entities from the current class model.
+ */
 + (NSArray *)all;
-+ (NSInteger)count;
+
+/**
+ * @brief Count all entities of the current class model.
+ *
+ * @return A NSInteger value corresponding to the total number of entities of the current class model.
+ */
+ + (NSInteger)count;
 
 #pragma mark - UPDATE
 
-- (void)updateWithDictionary:(NSDictionary *)dict;
+/**
+ * @brief Override to update the current entity with a given dictionary.
+ *
+ * @param dictionary A NSDictionary object containing information about the database entity to fetched with.
+ *
+ * @discussion Use this function and the given parameter to update the attributes of the current entity.
+ *
+ * Depending on your project and architecture the information/data of the child entities could be inside the dictionary.
+ * In this case initiate the CRUD process from this function.
+ *
+ * The super function should be called.
+ *
+ * For example, a `Book` entity could have some `Images` as child entities.
+ * @code
+ * super.updateWithDictionary(dictionary)
+ * self.id = GET_NUMBER(dict, "id")
+ * self.title = GET_STRING(dict, "title")
+ * // child entities
+ * self.images = Images.createEntitiesFromArray(GET_ARRAY(dict, "images"))
+ * @endcode
+ */
+- (void)updateWithDictionary:(NSDictionary *)dictionary;
 
 #pragma mark - DELETE
 
 - (BOOL)deleteIfInvalid;
+
+/**
+ * @brief Override this function to delete the child entities of a current entity.
+ *
+ * @discussion To remove an entity one should call `deleteEntityWithReason:`.
+ * When doing so the current function is called.
+ * The expected behavior is to remove any data store on the disk (like image assets)
+ * but also to forward the destruction process to its child entities.
+ * The super function should always be called.
+ *
+ * When implementing a DB that matches a distant database (behind an API) this function
+ * might also be called when an entity gets deprecated/disabled.
+ *
+ * The super function should be called.
```

Comment:
You already wrote that. Some lines above: `The super function should always be called.`